data-processing/common/upload_entities.py

```diff
@@ -309,3 +313,53 @@ def resolve_model(entity_identifier: EntityIdentifier) -> Optional[Entity]:
                     )
 
     return models
+
+
+def write_to_file(entity_infos: List[EntityUploadInfo], output_file_name: str) -> None:
+    # an attempt to make life easier if we change how we want to format this
+    FORMAT_VERSION = "v0"
+
+    logging.info(
+        "About to write %d entity infos to %s (version: %s).",
+        len(entity_infos),
+        output_file_name,
+        FORMAT_VERSION,
+    )
+    to_write = {
+        "version": FORMAT_VERSION,
+        "data": [dataclasses.asdict(entity_info) for entity_info in entity_infos],
+    }
+    already_exists_msg = (
+        f"File {output_file_name} already exists. "
+        + "Not overwriting. Entity infos will not be written."
+    )
```

Comment:
The default behavior in this repo is to overwrite while printing a warning message. We could do this behavior instead, and in that case I would be curious of your thinking about if this is a better way to do it, in case it makes sense for us to change the behavior elsewhere in the repo.

Reply:
I think I tend towards the better safe than sorry route, avoiding overwriting existing things, and placing the responsibility on the user to clean up before running things, if they want output to be stored in the same place.
But I don't feel particularly strongly about this with scholarphi, and I also think that especially with this kind of thing, where things can run for a really long time, it could be quite frustrating if a pipeline completes after a number of hours, only to let you know that you don't get to see any of the output because you forgot to clear the relevant directory or point it somewhere else. I've updated the behavior to print a warning message, then overwrite:
```
...
Saving to file...
2021-09-18 01:01:59,783 [INFO]: About to write 9 entity infos to hi/citations.jsonl (version: v0).
2021-09-18 01:01:59,783 [WARNING]: File hi/citations.jsonl already exists. It will be overwritten.
2021-09-18 01:01:59,785 [INFO]: Finished running command upload-citations
...
```